ldclient/streaming.py

```diff
@@ -30,8 +33,17 @@ def run(self):
                 for msg in messages:
                     if not self._running:
                         break
-                    if self.process_message(self._store, self._requester, msg, self._ready) is True:
+                    message_ok = self.process_message(self._store, self._requester, msg, self._ready)
+                    if message_ok is True and self._ready.is_set() is False:
                         self._ready.set()
+            except HTTPError as e:
+                if e.response is not None and e.response.status_code is not None:
```

Comment:
which begs the question- is it possible for a 404 error to be transient? due to a misconfigured/misbehaving CDN?